telecine/src/main/java/com/jakewharton/telecine/RecordingSession.java

```diff
@@ -395,4 +405,24 @@ public void destroy() {
       stopRecording();
     }
   }
+
+  private static int getNotificationId() {
+    return (int) SystemClock.uptimeMillis();
+  }
+
+  public static class DeleteRecordingBroadcastReceiver extends BroadcastReceiver {
+
+    @Override public void onReceive(Context context, Intent intent) {
+      int rowsDeleted = context.getContentResolver().delete(intent.getData(), null, null);
```

Comment:
My concern is if one video becomes too large, it may take some time to delete it. This query is handled by [MediaProvider](https://github.com/android/platform_packages_providers_mediaprovider/blob/master/src/com/android/providers/media/MediaProvider.java) but i did not find wether this is handled on a another  thread.